[DOC-366] Add Dagster Plus audit log guide (#23931)
dagster-io/dagster: docs/docs-beta/docs/dagster-plus/access/rbac/audit-logs.md

sidebar_position: 4
---

# Audit logs

> The Dagster Plus audit log enables enterprise organizations to track and attribute changes to their Dagster deployment.

Reviewer: Should we use the word `enterprise` here? Perhaps it be something that aligns with our pricing plans, ex:

> The Dagster+ audit log enables Dagster+ Pro organizations...

Reviewer: Also, maybe include a sentence or two about why having the ability to track and attribute deployment changes is helpful.

This guide walks through how to access the audit log and details the interactions which are tracked in the audit log.

<details>
<summary>Prerequisites</summary>
- A Dagster Plus Pro organization

> - An [Organization Admin](/dagster-plus/access/rbac/user-roles-permissions) role in your Dagster Plus organization

Reviewer: ```suggestion
- An [Organization Admin](/dagster-plus/access/rbac/user-roles-permissions) role in your Dagster+ organization
```

I won't clutter the review to point this out, but could you please do a find/replace for `Dagster Plus` to `Dagster+`?

</details>

## View audit logs

> Audit logs are available in the Organization settings pane, which can be accessed by clicking on your user icon on the top right side of the screen and selecting Organization settings in the dropdown menu. Once on the Organization settings pane, select Audit log on the top bar.

Reviewer: Since these are sequential steps, let's put them in a numbered list:

```suggestion
To access the audit logs:

1. Click your user icon at the top right corner of the page.
2. Click **Organization settings**.
3. Click the **Audit log** tab.
```

> 

Reviewer: Could you add a TODO for a screenshot here? I think it'd be good to show the logs.

> Each entry in the audit log indicates when an action was taken, the user who performed the action, the action taken, and the deployment which the action affected. You may click the "Show" button to view additional details for any audit log action.

Reviewer: ```suggestion
Each entry in the audit log indicates when an action was taken, the user who performed the action, the action taken, and the deployment which the action affected. To view additional details for an action, click the **Show** button.
```

Suggesting we flip the last sentence around to lead with what the button does.

## Filter the audit log

> The "Filter" button on the top left of the Audit log pane can be used to filter the list of logs. You can filter to a combination of user, event type, affected deployment, or time frame.

Reviewer: ```suggestion
The **Filter** button near the top left of the page can be used to filter the list of logs. You can filter to a combination of user, event type, affected deployment, or time frame.
```

- UI components should be bolded
- Suggesting we generally describe where the button is located on the page instead of relation to a (non-labeled) element

| Update agent token permissions | A user alters [permissions](/dagster-plus/access/rbac/user-roles-permissions) for an agent token             | The permission grant and targeted deployment                              |
| Create secret                  | A user creates a new [environment variable](/dagster-plus/deployment/environment-variables/dagster-ui)        | The created variable name                                                 |
| Update secret                  | A user modifies an existing [environment variable](/dagster-plus/deployment/environment-variables/dagster-ui) | The previous and current variable names and whether the value was changed |
| Delete secret                  | A user removes an [environment variable](/dagster-plus/deployment/environment-variables/dagster-ui)           | The deleted variable name                                                 |
| Update subscription            | A user modifies the selected Dagster Plus subscription for the organization                                   | The previous and current plan types                                       |

## Programmatic access to audit logs

> Audit logs can be accessed over the Dagster Plus GraphQL API to access them programmatically:

Reviewer: ```suggestion
Audit logs can be accessed programmatically over the Dagster+ GraphQL API:
```

The wording of this was a little bit awkward

 for tracking operational history, including sensor and schedule updates.

This guide walks through how to access the audit log and details the interactions which are tracked in the audit log.

<details>
<summary>Prerequisites</summary>
- A Dagster+ Pro organization
- An [Organization Admin](/dagster-plus/access/rbac/user-roles-permissions) role in your Dagster+ organization
</details>

## View audit logs

To access the audit logs:

1. Click your user icon at the top right corner of the page.
2. Click **Organization settings**.
3. Click the **Audit log** tab.

:::warning

Add screenshot

:::

Each entry in the audit log indicates when an action was taken, the user who performed the action, the action taken, and the deployment which the action affected. To view additional details for an action, click the **Show** button.

## Filter the audit log

The **Filter** button near the top left of the page can be used to filter the list of logs. You can filter to a combination of user, event type, affected deployment, or time frame.

## Audit log entry types

> | Event type                     | Description                                                                                                   | Additional details                                                        |

Reviewer 2: I love this table -- how often do we add audit log events and is there a risk it will become stale?

Reviewer: @OwenKephart's [auto-generating alert doc](examples/docs_beta_snippets/docs_beta_snippets/dagster-plus/deployment/alerts/generate_alerts_doc.py) makes me want to give similar treatment to everything like this. Could look into adapting it.

Author: That's a great question. Took a stab at making a parity unit test here which would give us some notification if the tables got out of sync: https://github.com/dagster-io/internal/pull/11224

Not sure if in practice this is more annoying than it's worth
